File: python/28d_1343B_BalancedArray.py
def main():
    t = int(input())
    n = []
    for _ in range(t):
        n.append(int(input()))
    for i in n:
        if i % 4 != 0:
            print("NO")
        else:
            print("YES")
            ans = []
            for j in range(0, (i // 2)):
                ans.append((2*(j+1)))
            for j in range(0, (i // 2) - 1):
                ans.append(ans[j] - 1)
            ans.append((3 * (i // 2)) - 1)
            for j in ans:
                print(j, end = " ")
            print()

if __name__ == "__main__":
    main()

File: cpp/28d_1343B_BalancedArray.cpp
#include<bits/stdc++.h>

int main(){
    int t, n;
    std::vector<int> ns;
    std::cin>>t;
    for(int i = 0; i < t; i++){
        std::cin>>n;
        ns.push_back(n);
    }
    for(auto i : ns){
        if (i % 4 != 0){
            std::cout<<"NO"<<"\n";
        }
        else{
            std::vector<int> ans;
            std::cout<<"YES"<<"\n";
            for(int j = 0; j < (i / 2); j++){
                ans.push_back((2*(j + 1)));
            }
            for(int j = 0; j < (i / 2) - 1; j++){
                ans.push_back(ans[j] - 1);
            }
            ans.push_back((3*(i / 2)) - 1);
            for(auto j : ans){
                std::cout<<j<<" ";
            }
            std::cout<<"\n";
        }
    }
    return 0;
}